realm/src/main/java/io/realm/internal/NativeObject.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal;
+
+/**
+ * This abstract class represents a native object from core.
+ * It specifies the operations common to all such objects.
+ * All Java classes wrapping a core class should extend NativeObject.
```

Comment:
Yes, and right now we plan to keep it this way, but it's nice to have the infrastructure ready to accommodate more classes if we ever need to.